### DISCUSSION TITLE
Provide an option to serve public files from CDN

### DISCUSSION THREAD

**Discussion by FulcronZ** [vercel/next.js#29707] [Ideas]
### Describe the feature you'd like to request

Public files have to be prefixed manually in pages to be served from CDN at the moment, 
despite of utilizing [`publicRuntimeConfig`][public_runtime]. 

In addition to serving `.next/static/`  from CDN with [`assetPrefix`][asset_prefix] in `next.config.js` option,
public files may get served from CDN too in similar fashion without prefixing manually in pages. 

[public_runtime]: https://nextjs.org/docs/api-reference/next.config.js/runtime-configuration
[asset_prefix]: https://nextjs.org/docs/api-reference/next.config.js/cdn-support-with-asset-prefix 

### Describe the solution you'd like

Provide a new `next.config.js` option to prefix public assets: `publicAssetPrefix`. 

### Describe alternatives you've considered

None

**Comment by Sphinm** [comment]
I support you. 

I just met this problem and don’t want to manually add the cdn prefix to the public. So I hope to add a default configuration.

**Comment by wangping51** [comment]
这个问题解决了吗？
